### PR TITLE
IMPRO-1507: A CLI to apply reliability calibration

### PR DESCRIPTION
**Pull request by bayliffe**
A new CLI and acceptance test for the application of reliability calibration.

Testing:
 - [x] Ran tests and they passed OK
 - [x] Added new tests for the new feature(s)

### CODE REVIEW

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- improver/cli/apply_reliability_calibration.py  1
- improver_tests/acceptance/test_apply_reliability_calibration.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5301076&bid=17351244)

File: improver/cli/apply_reliability_calibration.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            minimum_forecast_count=200):
+    """Apply reliability calibration.
+
+    Calibrate a forecast using the provided reliability calibration table.
+
+    Args:
+        forecast (iris.cube.Cube):
+            The forecast to be calibrated.
+        reliability_table (iris.cube.Cube):
+            The reliability calibration table to use in calibrationg the
```
Comment (gavinevans) [inline improver/cli/apply_reliability_calibration.py:51]: Typo: `calibrationg` => `calibrating`

File: improver/cli/apply_reliability_calibration.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+@cli.clizefy
+@cli.with_output
+def process(forecast: cli.inputcube,
+            reliability_table: cli.inputcube,
+            *,
+            minimum_forecast_count=200):
+    """Apply reliability calibration.
+
+    Calibrate a forecast using the provided reliability calibration table.
```
Comment (gavinevans) [inline improver/cli/apply_reliability_calibration.py:45]: Would it be worth saying a bit more about what "apply reliability calibration" means and/or referencing Jonathan's paper, just to give a bit more context?

File: improver/cli/apply_reliability_calibration.py (hunk 1 above)
Comment (bayliffe) [reply improver/cli/apply_reliability_calibration.py:51]: Fixed.

File: improver/cli/apply_reliability_calibration.py (hunk 2 above)
Comment (bayliffe) [reply improver/cli/apply_reliability_calibration.py:45]: Done.

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- improver/cli/apply_reliability_calibration.py  1
- improver_tests/acceptance/test_apply_reliability_calibration.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5301076&bid=17351244)

File: improver/cli/apply_reliability_calibration.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+    Args:
```
Comment (fionaRust) [inline improver/cli/apply_reliability_calibration.py:47]: Missing minimum_forecast_count arg in docstring

File: improver_tests/acceptance/test_apply_reliability_calibration.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+pytestmark = [pytest.mark.acc, acc.skip_if_kgo_missing]
+CLI = acc.cli_name_with_dashes(__file__)
+run_cli = acc.run_cli(CLI)
+
+
+def test_calibration(tmp_path):
+    """
+    Test calibration of a forecast using a reliability calibration table.
+    """
+    kgo_dir = acc.kgo_root() / "apply-reliability-calibration/basic"
```
Comment (fionaRust) [inline improver_tests/acceptance/test_apply_reliability_calibration.py:46]: Is the the same as you used in this notebook? https://github.com/MetOffice/improver_aux/blob/master/notebooks/reliability_calibration/IMPRO-1506_application_of_reliability_calibration.ipynb

Comment (bayliffe) [reply improver_tests/acceptance/test_apply_reliability_calibration.py:46]: Yes, sorry I should have written that somewhere.

File: improver/cli/apply_reliability_calibration.py (hunk 2 above)
Comment (bayliffe) [reply improver/cli/apply_reliability_calibration.py:45]: I made this change in the wrong place. Oops. Adding more detail here now @gavinevans . Sorry.

File: improver/cli/apply_reliability_calibration.py (hunk 3 above)
Comment (bayliffe) [reply improver/cli/apply_reliability_calibration.py:47]: Added.

Comment (bayliffe) [reply improver/cli/apply_reliability_calibration.py:47]: Added.

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- improver/cli/apply_reliability_calibration.py  1
- improver_tests/acceptance/test_apply_reliability_calibration.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5301076&bid=17351244)

**Pull request merged**